app/api/routers/breeze_buddy/templates/handlers.py

```
@@ -297,3 +301,106 @@ async def get_template_by_id_handler(template_id: str, current_user: UserInfo):
+        f"User {current_user.username} (role: {current_user.role}) updating template "
+        f"{template_id}"
+    )
+
+    try:
+        # Check if template exists
+        existing_template = await get_template_by_id(template_id)
+        if not existing_template:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"Template not found: {template_id}",
+            )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)